src/main/java/lotto/model/LottoResult.java

```diff
@@ -35,53 +34,66 @@ private void calculateLottoResult() {
   }
 
   private void calculateLottoMatch(Lotto lotto) {
-    int matchingCount = lotto.getMatchingCount(winningLotto.getLotto());
+    LottoRank lottoRank = LottoRank.getLottoRank(
+        lotto.getMatchingCount(winningLotto.getLotto()),
+        lotto.isContainNumber(winningLotto.getBonusNumber())
+    );
```

Comment:
```suggestion
    LottoRank lottoRank = winningLotto.rank(lotto);
```
winningLotto에 메시지를 던져서 등수를 리턴받는건 어떤가요?🤔

Reply:
WinningLotto 필드에서 관리하는게 아니라, 메세지를 던졌을 때 리턴 받도록 하는걸 말씀하시는게 맞으실까요 ??

Reply:
넵 반영했습니다.
winningLotto에 메시지를 던져서 등수를 리턴을 받도록 수정을 하였습니다.
근데 한편으로는 winningLotto 에게 LottoRank 를 묻는게 맞는지 확신이 안서는데요.
WInningLotto 가 LottoRank 결과까지 반환하는 기능을 가지고 있어도 괜찮은지, 아니면 WinningLotto 는 LottoRank 까지 관여하지 않아야 하는지 조금 헷갈리는데요.
근환님은 어떤 근거로 winningLotto 에게 해당 메세지를 던져 리턴 받으려고 했는지 알려주시면 감사하겠습니다.
제가 객체 지향 적인 사고에 미숙하여서 답변주시면 감사하겠습니다.

Reply:
`등수`를 반환하는 역할이 `winningLotto` 외부에 있다면
`등수`를 반환하기 위해 `winningLotto`의 구현에 의존적이 될수밖에 없다고 생각합니다. (로또번호와 보너스번호를 gettter를 통해서 얻어와서 계산하는 등)🤔
만약 `winningLotto`가 변경된다면 외부의 등수반환 로직도 변경해아할 가능성이 높죠.

`winningLotto`가 `등수`를 반환한다면 `winningLotto`를 의존하고 있는 클라이언트는 `winningLotto`의 내부 구현에 대한 의존성을 덜게 됩니다.
반면 `winningLotto`가 `LottoRank`에 대한 의존성을 갖게되죠.

저는 `winningLotto`와 `LottoRank`가 관련성이 높은 관계라고 생각했고, 변경의 주기가 같을것이라고 생각해서 위와 같이 코멘트를 드렸어요.😊

객체지향 설계에 정답은 없으니 상황에 따른 선택을 하시는게 중요할것 같아요!👍